src/ripple/app/hook/applyHook.h

```diff
@@ -888,12 +1288,396 @@ class HookExecutor
+
+            return retval;
+        }
+
+        JS_FreeValue(ctx, val);
+
+        const char* testCalls = "if (typeof(Hook) != \"function\" || "
+            "(typeof(Callback) != \"function\" && typeof(Callback) != \"undefined\")) "
+            "throw Error(\"Hook/Callback function required\")";
+
+        val =
+            JS_Eval(vm.ctx, testCalls, sizeof(testCalls)-1, "<qjsvm>", 0);
+
+        if (JS_IsException(val))
+        {
+            if (const char* str = JS_ToCString(ctx, val); str)
+            {
+                retval.emplace(str);
+                JS_FreeCString(ctx, str);
+            }
+        }
+
+        JS_FreeValue(ctx, val);
+        //JS_FreeValue(ctx, obj);
+
+        return retval;
+    }
+
+    /**
+     * Execute QuickJS bytecode against hook context.
+     * Once execution has occured the exector is spent and cannot be used again
+     * and should be destructed Information about the execution is populated
+     * into hookCtx
+     */
+    void
+    execute(
+        const void* buf,
+        size_t buf_len,
+        bool callback,
+        uint32_t hookArgument,
+        uint32_t instructionLimit,
+        beast::Journal const& j) override
+    {
+        // HookExecutorWasm can only execute once
+        assert(!spent);
+        spent = true;
+
+        JLOG(j.trace()) << "HookInfo[" << HC_ACC()
+                        << "]: creating quickjs instance";
+
+        QuickJSVM vm{reinterpret_cast<void*>(&hookCtx), instructionLimit};
+        JSContext* ctx = vm.ctx;
+
+        if (!vm.sane())
+        {
+            JLOG(j.warn()) << "HookError[" << HC_ACC()
+                           << "]: Could not create QUICKJS instance.";
+
+            hookCtx.result.exitType = hook_api::ExitType::JSVM_ERROR;
+            return;
+        }
+        
+        JSValue obj =
+            JS_ReadObject(ctx, (uint8_t const*)buf, buf_len, JS_READ_OBJ_BYTECODE);
+        
+        if (JS_IsException(obj))
+        {
+            JS_FreeValue(ctx, obj);
+            JLOG(j.warn()) << "HookError[" << HC_ACC()
+                           << "]: Could not create QUICKJS instance (invalid bytecode).";
+            hookCtx.result.exitType = hook_api::ExitType::JSVM_ERROR;
+            return;
+        }
+
+        JSValue val =
+            JS_EvalFunction(ctx, obj);
+
+        if (JS_IsException(val))
+        {
+            JS_FreeValue(ctx, val);
+            JS_FreeValue(ctx, obj);
+
+            JLOG(j.warn()) << "HookError[" << HC_ACC()
+                           << "]: Could not create QUICKJS instance (bytecode eval failure).";
+            hookCtx.result.exitType = hook_api::ExitType::JSVM_ERROR;
+            return;
+        }
+
+        JS_FreeValue(ctx, val);
+
+        char expr[256];
+
+        int expr_len = 
+            snprintf(expr, 256, "%s(%d)",
+                callback ? "Callback" : "Hook",
+                hookArgument);
+
+        if (expr_len < 7 || expr_len == 256)
```

Comment:
This change makes it possible to do #332 Request 2.

```suggestion
        int expr_len = 
            snprintf(expr, 256, "%s(%d, otxn_json())",
                callback ? "Callback" : "Hook",
                hookArgument);

        if (expr_len < 20 || expr_len == 256)
```